packages/sanity/src/core/store/bundles/useBundleOperations.ts

```diff
@@ -59,19 +60,42 @@ export function useBundleOperations() {
         .filter(([_, value]) => value === undefined)
         .map(([key]) => key)
 
-      let clientOperation = client.patch(bundle._id).set(document)
+      let clientOperation = addOnClient.patch(bundle._id).set(document)
       if (unsetKeys.length) {
         clientOperation = clientOperation.unset(unsetKeys)
       }
 
       return clientOperation.commit()
     },
-    [client],
+    [addOnClient],
+  )
+
+  const handlePublishBundle = useCallback(
+    async (bundleId: string, bundleDocuments: SanityDocument[]) => {
+      if (!addOnClient) return null
```

Comment:
I know we are handling this in the same case in the rest of operations, but I'm thinking if maybe instead we should throw an error in the case that addOnClient is undefined.
Could also be a follow up.

Reply:
@pedrobonamin yeh good idea - added a throw to all the methods here